core/src/main/java/hudson/model/listeners/RunListener.java

```diff
@@ -202,43 +202,34 @@ public void unregister() {
      * Fires the {@link #onCompleted(Run, TaskListener)} event.
      */
     public static void fireCompleted(Run r, @NonNull TaskListener listener) {
-        for (RunListener l : all()) {
-            if(l.targetType.isInstance(r))
-                try {
-                    l.onCompleted(r,listener);
-                } catch (Throwable e) {
-                    report(e);
-                }
-        }
+        Listeners.notify(RunListener.class, l -> {
+            if (l.targetType.isInstance(r)) {
```

Comment:
This repeated bit of logic could I guess be encapsulated in a new `static` helper method here. Maybe not worth the bother.

Reply:
Could indeed, but I have to draw the line somewhere so I won't bother to fix this pre-existing issue.